### DISCUSSION TITLE
c7n: S3 actions to enable S3 server access logging,  set target bucket and and set target prefix

### DISCUSSION THREAD

**Discussion by epluscloudservices** [cloud-custodian/cloud-custodian#7332] [General]
This is a feature request for c7n `s3 resource actions` that can:

1) enable server access logging
2) set target bucket
3) set target prefix using the name of the bucket

The use case is for security audits and compliance.

**Comment by kapilt** [comment]
`toggle-logging` is the action name for this, and supports variable interpolation around the prefix. fwiw, access logging is a bit older wrt to current feature set, i'd suggest looking at s3 data events for this use case as the logging is significantly richer

**Comment by davidclin** [comment]
That worked flawlessly.  Thank you!!!

**Comment by cvemula1** [comment]
Is im doing something wrong here ? I'm trying to enable access logging specific to single bucket     
resource: s3
    filters:
      - and:
        - type: is-log-target
          services:
            - cloudtrail
        - type: value
          key: "Logging.TargetBucket"
          value: absent
        - type: data-events
          state: absent
        - type: value
          key: trail
          value: GovernanceCloudTrail
    actions:
      - type: toggle-logging
